clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/AdaptiveThrottlerSignalHub.java

```diff
@@ -0,0 +1,67 @@
+package com.linkedin.davinci.kafka.consumer;
+
+import com.linkedin.davinci.stats.ingestion.heartbeat.AggregatedHeartbeatLagEntry;
+import com.linkedin.davinci.stats.ingestion.heartbeat.HeartbeatMonitoringService;
+import io.tehuti.metrics.MetricsRepository;
+
+
+/**
+ * This class contains host-level signals for adaptive ingestion throttling logic.
+ */
+public class AdaptiveThrottlerSignalHub {
+  private final MetricsRepository metricsRepository;
+  private final HeartbeatMonitoringService heartbeatMonitoringService;
+
+  private boolean singleGetLatencySignal = false;
+  private boolean currentLeaderMaxHeartbeatLagSignal = false;
+  private boolean currentFollowerMaxHeartbeatLagSignal = false;
+  private boolean nonCurrentLeaderMaxHeartbeatLagSignal = false;
+  private boolean nonCurrentFollowerMaxHeartbeatLagSignal = false;
+
+  public AdaptiveThrottlerSignalHub(
+      MetricsRepository metricsRepository,
+      HeartbeatMonitoringService heartbeatMonitoringService) {
+    this.metricsRepository = metricsRepository;
+    this.heartbeatMonitoringService = heartbeatMonitoringService;
+  }
+
+  public void refreshSignal() {
```

Comment:
how do you plan to refresh it? have another background thread to refresh?

Reply:
I didn't think about this part in the first iteration, actually I can add the thread and refresh all the registered throttler.